tests/test_worker.py

```diff
@@ -213,5 +215,36 @@ def test_worker_address(hq_env: HqEnv):
     assert output == gethostname()
 
 
+@pytest.mark.parametrize("policy", ["stop", "finish-running"])
+def test_server_lost_no_tasks(hq_env: HqEnv, policy: str):
+    hq_env.start_server()
+    worker = hq_env.start_worker(args=[f"--on-server-lost={policy}"])
+    hq_env.kill_server()
+    time.sleep(0.5)
+    hq_env.check_process_exited(worker)
+
+
+def test_server_lost_stop_with_task(hq_env: HqEnv):
+    hq_env.start_server()
+    worker = hq_env.start_worker(args=["--on-server-lost=stop"])
+    hq_env.command(["submit", "--array=1-10", "--", "bash", "-c", "sleep 2"])
+    wait_for_job_state(hq_env, 1, "RUNNING")
+    hq_env.kill_server()
+    time.sleep(0.5)
+    hq_env.check_process_exited(worker)
+
+
+def test_server_lost_finish_running_with_task(hq_env: HqEnv):
+    hq_env.start_server()
+    worker = hq_env.start_worker(args=["--on-server-lost=finish-running"])
+    hq_env.command(["submit", "--array=1-10", "--", "bash", "-c", "sleep 2"])
+    wait_for_job_state(hq_env, 1, "RUNNING")
+    hq_env.kill_server()
+    time.sleep(0.5)
```

Comment:
Maybe we should check that the task was finished, e.g. some file could be written to disk?